test/sql/storage/compression/zstd/page_flushing_test1.test

```diff
@@ -0,0 +1,112 @@
+# name: test/sql/storage/compression/zstd/page_flushing_test1.test
+# group: [zstd]
+
+load __TEST_DIR__/zstd_page_flushing_test1.db
+
+require block_size 262144
+
+require vector_size 2048
```

Comment:
Can we add some more tests?

* Tests with skipping, e.g. we have a column `id` that has repeated numbers `1 x 5000, 2 x 5000, etc` and a zstd compressed column - we then read from the table using a filter (e.g. `id > 1`, `id > 2`, etc)
* Tests with bigger column segments, this is in particular the case when have a `VARCHAR[]` with multiple string values per list. This can also mean we scan unaligned vectors, particularly if the string offsets are odd

Reply:
> This can also mean we scan unaligned vectors, particularly if the string offsets are odd

I'm not sure I follow this part?
The big string lists resulted in implementing support for producing multiple segments (if the vector metadata wont all fit on the first page)